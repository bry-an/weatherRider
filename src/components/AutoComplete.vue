<template>
  <div>
    <label for="origin">Origin</label>
    <input type="text" ref="origin" v-model="originInput" placeholder="Enter Start Location">
    <label for="destination">Destination</label>
    <input
      type="text"
      ref="destination"
      v-model="destinationInput"
      placeholder="Enter End Location"
    >

    <!-- <button @click="setOrigin">Set origin</button> -->
  </div>
</template>
<script>
export default {
  name: "AutoComplete",
  data() {
    return {
      originInput: "",
      destinationInput: "",
      place: null,
      google: window.google
    };
  },
  methods: {
    autoComplete(ref) {
      const googleMaps = this.google.maps;
      const autocomplete = new googleMaps.places.Autocomplete(ref);
      autocomplete.setFields(["geometry"]);
      const infowindow = new this.google.maps.InfoWindow();
      autocomplete.addListener("place_changed", () => {
        infowindow.close();
        let placeObject = autocomplete.getPlace();
        console.log(placeObject);
        this.place = placeObject.geometry.location;
        this.$store.commit("setMapCenter", {
          lat: this.place.lat(),
          lng: this.place.lng()
        });
      });
    },
    setOrigin() {
      this.$store.commit("setOrigin", this.place);
    },
    log() {
      console.log("placeinput in component", this.placeInput);
    }
  },
  mounted() {
    this.autoComplete(this.$refs.origin);
    this.autoComplete(this.$refs.destination);
  }
};
</script>



