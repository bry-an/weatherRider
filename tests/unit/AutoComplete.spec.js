import { shallowMount } from "@vue/test-utils";
import AutoComplete from "@/components/AutoComplete.vue";
import googleObject from "../fixtures/googleObject.js";

describe("AutoComplete", () => {
  const build = () => {
    const wrapper = shallowMount(AutoComplete);
    wrapper.setData({
      google: googleObject,
      placeInput: "hello"
    });
    return {
      wrapper
    };
  };

  it("displays correctly", () => {
    const { wrapper } = build();
    console.info(wrapper.vm.placeInput);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
