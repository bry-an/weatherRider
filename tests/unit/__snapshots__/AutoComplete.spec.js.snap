// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoComplete displays correctly 1`] = `<div><input type="text" placeholder="Enter Location"></div>`;
